docs/docs/reading-events/Serverside filtering.txt

```diff
@@ -0,0 +1,210 @@
+other-event @ 3029
+```
+
+More often than not you won't care about the `$` system events so we can filter those out by adding a filter to your subscription.
+
+```
+await client.SubscribeToAllAsync(Position.Start,
+    (s, e, c) =>
+    {
+        Console.WriteLine($"{e.Event.EventType} @ {e.Event.Position.PreparePosition}");
+        return Task.CompletedTask;
+    },
+    filterOptions: new SubscriptionFilterOptions(EventTypeFilter.ExcludeSystemEvents())
+);
+```
+
+And if you run it again you'll now get the following output.
+
+```
+some-event @ 1540
+other-event @ 1705
+some-event @ 1871
+other-event @ 2036
+some-event @ 2202
+other-event @ 2367
+some-event @ 2533
+other-event @ 2698
+some-event @ 2864
+other-event @ 3029
+```
+
+But what if you only want to listen for `some-event` and ignore all the other events including the `other-event`? Well in this case you can pass in a prefix filter.
+
+```
+new SubscriptionFilterOptions(EventTypeFilter.Prefix("some-"))
+```
+
+To get the following output.
+
+```
+some-event @ 1540
+some-event @ 1871
+some-event @ 2202
+some-event @ 2533
+some-event @ 2864
+```
+
+As mentioned above you can also filter by regular expressions and stream names. Following is a few more examples that may be useful.
+
+```
+new SubscriptionFilterOptions(EventTypeFilter.RegularExpression(@"some|other") // will filter for event types containing the words "some" or "other"
+
+new SubscriptionFilterOptions(StreamFilter.Prefix("user")) // will filter for streams beginning with "user"
+
+new SubscriptionFilterOptions(StreamFilter.RegularExpression("^user|^company") // Will only return events beginning with "user" or "company"
+```
+
+As you can see this will hopefully replace the use case for projections in a lot of scenarios, which in turn will reduce write amplification and improve the performance of EventStoreDB.
+
+## Checkpointing
+There is one thing to consider with this solution and that is when events that match your filter are few and far between. In this scenario you might find yourself in the situation where EventStoreDB has searched through 1million events and the last thing you want to happen is for the server to get to event 900k and then have your client crash. It won't have been able to take a checkpoint and upon restart up you'd have to go all the way back to the beginning and start again.
+
+It's for this reason we have introduced an additional delegate that will be triggered every n number of events.
+
+To make use of it setup the extra delegate `checkpointReached` on the SubscriptionFilterOptions class.
+
+```
+var filterOptions = new SubscriptionFilterOptions(
+		EventTypeFilter.ExcludeSystemEvents(),
+    checkpointReached: (s, p, c) =>
+    {
+        Console.WriteLine($"checkpoint taken at {p.PreparePosition}");
+        return Task.CompletedTask;
+    });
+```
+
+ This will be called every n number of events. I you want to be specific about the number of events threshold you can also pass that in.
```

Comment:
```suggestion
 This will be called after every n number of events have been traversed on the server. I you want to be specific about the number of events threshold you can also pass that in.
```